tabs/pid_tuning.js

```diff
@@ -498,3 +531,171 @@ TABS.pid_tuning.cleanup = function (callback) {
         callback();
     }
 };
+
+/* Wrap whole function in an independant class */
+
+function ExpoChart(canvas, rcData, rcExpo, rcRate, deadband, midrc, axisRate, superExpoActive) {
```

Comment:
You should move these parameters (except for the canvas) into the `refresh` function. This way you can update the curve without recreating the object.